src/server/src/server_main.cpp

```diff
@@ -0,0 +1,69 @@
+//
+// Created by kevin on 6/25/22.
+//
+#include <ros/ros.h>
+#include <client/current_time.h>
+#include <client/login.h>
+#include <vector>
+
+#include <iostream>
+
+ros::NodeHandle *server_node_handle_pointer = NULL;
+ros::ServiceServer service_server;
+std::vector<ros::Subscriber> server_subscribers;
+
+void time_callback(const client::current_time::ConstPtr &msg) {
+    ROS_INFO("Topic received from : %s , Whose time is %d/%d/%d-%d:%d:%d ."
+            , (msg->name).c_str()
+            , msg->year
+            , msg->month
+            , msg->day
+            , msg->hour
+            , msg->minute
+            , msg->second
+    ); //Output
+}
+
+bool login_handle(client::login::Request &req, client::login::Response &ack) {
+    if(req.req_code == 1) {
+        ROS_INFO("A client: %s has logged in!", req.node_name.c_str());
+        ack.ack_code = 10;
+        server_subscribers.push_back(
+                (*server_node_handle_pointer).subscribe(
+                        req.node_name, 1, time_callback));
+    }else{
+        ROS_INFO("A client: %s has logged out!", 
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]